File: week5/day1/getSquares.js
console.log('square');

var getSquares = function (array) {
    squareNum = []
    for(var i = 0; i < array.length; i++){
        if (array[i] % Math.sqrt(array[i]) === 0) {
            squareNum.push(array[i]);
        } 
    }
    var sortNum = squareNum.sort((a, b) => (a > b) ? 1 : -1);

    var uniqueNum = [...new Set(sortNum)];
    return uniqueNum
}

console.log(getSquares([4, 1, 16, 1, 10, 25, 22]));

File: week05/day1/get_squares.rb
# resolved with use of no methods

# get_squares = [4, 1, 16, 1, 10, 25, 22]
# square_nums = []

# get_squares.each do |num|
#     if (Math.sqrt(num) % 1).zero?
#         square_nums << num
#     end
# end

# p square_nums.sort.uniq


# method defined

def get_squares array
    square_nums = []
    array.each do |num|
        if (Math.sqrt(num) % 1).zero?
            square_nums << num
        end
    end
    p square_nums.sort.uniq
end

get_squares([4, 1, 16, 1, 10, 25, 22])